scripts/markdown/check-markdown.js

```diff
@@ -114,17 +110,14 @@ function checkMarkdownFiles(folderPath, validateFunction, relativePath = '') {
                 return;
             }
 
-            fs.stat(filePath, (err, stats) => {
-                if (err) {
-                    console.error('Error reading file stats:', err);
-                    return;
-                }
+            try {
```

Comment:
Why there are nesting try/catch in this function?

Reply:
the outer block manages directory-level issues like reading or iterating through folders, while the inner block handles file-specific errors such as reading markdown files

Reply:
But a single try catch can handle multiple errors inside that block, so you should have a single try catch block only.

Reply:
UPDATED